src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -155,10 +184,32 @@ private static String getBanDurationBreakdown(final Timestamp stamp) {
     return new BadWordController().list();
   }
 
-  private static String validatePassword(final Map<String, String> propertiesReadFromClient, final String clientName) {
+  private static String validatePassword(final Map<String, String> propertiesSentToClient,
+      final Map<String, String> propertiesReadFromClient, final String clientName) {
+    final String errorMessage = "Incorrect username or password";
     final UserDao userDao = new DbUserController();
+    final HashedPassword hashedPassword = userDao.getPassword(clientName);
+    if (hashedPassword == null) {
+      return errorMessage;
+    }
+    final String base64 = propertiesReadFromClient.get(ENCRYPTED_PASSWORD_KEY);
+    if (base64 != null) {
+      return decryptPassword(base64, propertiesSentToClient.get(RSA_PUBLIC_KEY), pass -> {
```

Comment:
There's no need to use the actual password, that's why we hash it use SHA512 and encrypt this hash which is then sent to the server, decrypted there and then hashed with bcrypt and stored in the database.